webview_panels/src/modules/queryPanel/components/perspective/PerspectivePlugins.js

```diff
@@ -0,0 +1,174 @@
+/* eslint-disable */
+import { getGlobalTheme } from "./PerspectiveViewer";
+
+// Dispatches a custom event with the given event name and message
+function dispatchCustomEvent(eventName, message, columnName, type) {
+  const event = new CustomEvent(eventName, {
+    detail: { columnName, message, type },
+  });
+  window.dispatchEvent(event);
+}
+
+// Checks if the given string is a valid JSON
+const isJson = (str) => {
+  try {
+    JSON.parse(str);
+  } catch (e) {
+    return false;
+  }
+  return true;
+};
+
+// Appends an image to the td element
+function appendImage(td) {
+  if (!td.querySelector("span")) {
+    const div = document.createElement("div");
+    const text = document.createElement("div");
+
+    // Setting text content to a div element to avoid base perspective viewer styling
+    text.textContent = td.textContent;
+    td.textContent = "";
+
+    // Adding svg icon to the span element
+    const span = document.createElement("span");
+    const img = document.createElement("img");
+
+    // Getting the current theme and setting the icon color
+    const currentTheme = getGlobalTheme();
+    const iconColor = currentTheme === "dark" ? "white" : "black";
```

Comment:
we should be able to use theme based css values without actual theme and also white and black are not right values and will look odd in different themes

Reply:
This is the only way I was able to render the svg there like I already showed you. Adding theme based icons is the only way to dynamically change it. white and black works correctly as seen in the screenshots above.